customer-commons/src/main/java/no/unit/nva/customer/model/CustomerDb.java

```diff
@@ -197,86 +204,110 @@ public void setVocabularySettings(Set<VocabularySettingDb> vocabularySettings) {
         this.vocabularySettings = vocabularySettings;
     }
 
+    public CustomerDto toCustomerDto() {
+        CustomerDto customerDto = CustomerDto.builder()
+            .withCname(this.getCname())
+            .withName(getName())
+            .withIdentifier(this.getIdentifier())
+            .withArchiveName(this.getArchiveName())
+            .withCreatedDate(this.getCreatedDate())
+            .withDisplayName(this.getDisplayName())
+            .withInstitutionDns(this.getInstitutionDns())
+            .withShortName(this.getShortName())
+            .withVocabularySettings(extractVocabularySettings())
+            .withModifiedDate(getModifiedDate())
+            .withFeideOrganizationId(getFeideOrganizationId())
+            .withCristinId(getCristinId())
+            .build();
+        return CustomerMapper.addContext(customerDto);
+    }
+
+    private static Set<VocabularySettingDb> extractVocabularySettings(CustomerDto dto) {
+        return Optional.ofNullable(dto.getVocabularySettings())
+            .stream()
+            .flatMap(Collection::stream)
+            .map(VocabularySettingDb::fromVocabularySettingsDto)
+            .collect(Collectors.toSet());
+    }
+
+    private Set<VocabularySettingDto> extractVocabularySettings() {
+        return Optional.ofNullable(this.getVocabularySettings())
+            .stream()
+            .flatMap(Collection::stream)
+            .map(VocabularySettingDb::toVocabularySettingsDto)
+            .collect(Collectors.toSet());
+    }
 
     public static final class Builder {
-        private UUID identifier;
-        private Instant createdDate;
-        private Instant modifiedDate;
-        private String name;
-        private String displayName;
-        private String shortName;
-        private String archiveName;
-        private String cname;
-        private String institutionDns;
-        private String feideOrganizationId;
-        private String cristinId;
-        private Set<VocabularySettingDb> vocabularySettings;
+
+        private final CustomerDb customerDb;
```

Comment:
Using an object directly allows us to define the logic of setting values (e.g. setVocabularySettings) only once and have better control of the set values.